### PR TITLE
new indicators and buffer methodology;

### PR DESCRIPTION
**Pull request by dcarver1**
Quite a few things in here .

indicator functions 
- these follow a very similar structure to the first few. Please eval for run completion within the main script and review code for any obvious errors in interpretation. 
functions/indicatorScoreCalculation/asthma.R
        new file:   functions/indicatorScoreCalculation/drought.R
        new file:   functions/indicatorScoreCalculation/heatDays.R
        new file:   functions/indicatorScoreCalculation/lowBirthWeight.R
- New method based on the EJScreen buffering process 
        new file:   functions/indicatorScoreCalculation/mining.R

There is a rmd file in the tutorial that explains the connection to all the functions/script required to produce the results. This is a new method and there are multiple steps, please test steps independently and provide on concerns or discrepencies with the application of the code. 

I don't have an exact answer on how population is applied to this buffer score caculation, which is why there are two outputs. TotalPopScore, ProprotionalPopScore 

We are meeting with EPA staff on the 3rd to get their impression on the work. Your welcome to join the meeting if that is interesting to you. 

Happy to talk through any of this work.

### CODE REVIEW

File: utilities/calculateDistanceScore.R [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  ## want this in km rather than meters for the 1/distance measures
+  blockSelection$distance <- round(distance[,1]) / 1000 
+  # add the site score from the site 
+  blockSelection$siteScore <- site$siteScore
+  
+  blockSelectionDistanceFilter <- blockSelection |>
+    dplyr::filter(distance <= 10)|> # this might change 
+    # add a measure for the very small distances, keeps the distance score to a max of 10 
+    dplyr::mutate(adjDistance = case_when( 
+      distance < 0.1 ~ 0.1,
```
Comment (ccmothes) [inline utilities/calculateDistanceScore.R:42]: So this is treating everything less than 0.1 km away the same?

File: functions/indicatorScoreCalculation/mining.R [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    }else{
+      allMining$cbg_geoid[i] <- index
+    }
+  }
+    
+  # remove any na values 
+  allMining_clean <- allMining |> dplyr::filter(!is.na(cbg_geoid))
+  
+  # define the site score value (different for each indicator. )
+  allMining_clean$siteScore <- 1
```
Comment (ccmothes) [inline functions/indicatorScoreCalculation/mining.R:129]: Why is the site score 1?

**Review by ccmothes — APPROVED** [review]
I went through all individual files/functions and noted where things did not work and added some general comments/questions that came up as in-line comments throughout. 

Other General Comments:

- Before sending code off for review, please test files on your end first (from a fresh R session). For example, there have been typos in multiple cases that prevent me from running the code that should be easy catches when running them on your end.

- Reading in functions in main - Unless I'm missing something here.. I think you could clean this up by dropping the need for the `loadFunctions` script and instead just load everything with a purr::map() and `source()` call. You could put utilities in a subfolder of functions and source them all at once?

File: 0_main.R [hunk 3]
```diff
@@ -34,8 +34,6 @@ pullCensusGeographies(overwrite = FALSE)
 ### generate a named list of all geometry objects used in the workflow
 geometries <- processGeometryLayers()
 
-
-
 # prepping for analysis  --------------------------------------------------
 ## adjust block populations
```
[the remark this answers is not in the document]
Comment (dcarver1) [reply 0_main.R:38]: done

File: 0_main.R [hunk 4]
```diff
@@ -63,15 +61,34 @@ plan(multisession, workers = 3)
...
+# environmental Effects ---- 
+getMining(geometryLayers = geometries)
+
+# climate vulnerability ----
+### wildfire
 getWildfire(filePath = "data/raw/wildfireRisk/Data/whp2023_GeoTIF/whp2023_cnt_conus.tif",
             geometryLayers = geometries)
-## sensitive populations ----
+### drought 
+getDrought(filePath = "data/raw/drought/dm_export_20190101_20231231.csv",
```
[the remark this answers is not in the document]
Comment (dcarver1) [reply 0_main.R:75]: will keep 
geometryLayers
and 
filePath / folderPath depending on the location path is point too

File: utilities/calculateDistanceScore.R [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+#' @param index : a numerical vector used to select rows from the sites dataframe 
+#' @param sites : a spatial data object with all point source emission locations
+#' @param blockGroupNeighbors : a dataframe with indexed for what block to evaluated 
+#' @param blocks : a spatial data object representation the block location 
+#'
+#' @return : a dataframe of block elements with a distance - populaiton weighted scores per the specific emission source  
+calculateDistanceScore <- function(index, sites, blockGroupNeighbors, blocks){
+  # select element of interest 
+  site <- data[index, ]
+  if(25 %% index){
```
[the remark this answers is not in the document]
Comment (dcarver1) [reply utilities/calculateDistanceScore.R:12]: it's a counter. 

When the index value is divisable by 25 it'll print 
We have a >1000 sites to loop over so this keeps some reference with out the massive print out

[the remark this answers is not in the document]
Comment (dcarver1) [reply utilities/calculateDistanceScore.R:12]: edited to include a conditional

File: functions/indicatorScoreCalculation/mining.R [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+  # block group relations 
+  blockGroupNeighbors <- readRDS("data/processed/geographies/bgNeighbors.RDS")
+  print("reading in reference layers")
+  # define the index for the map function
+  index <- 1:nrow(data)
+
+  # call the calculate score function 
+  exportFile <-  "data/products/environmentalEffects/mining/detailsOnDistanceScoring.csv"
+  if(!file.exists(exportFile)){
+    scores <- purrr::map(.x = index,
```
[the remark this answers is not in the document]
Comment (dcarver1) [reply functions/indicatorScoreCalculation/mining.R:17]: Switched to a for loop for this section. Need to look into how to pass the index into the purrr function. I think imap would do it

File: utilities/calculateDistanceScore.R (hunk 1 above)
Comment (dcarver1) [reply utilities/calculateDistanceScore.R:42]: yes, as per the documention the highest value the 1/distance should produce it 10. 

so anything less then 0.1 is converted to 0.1. Seemed easier route to ensure the behavior

File: 0_main.R [hunk 7]
```diff
@@ -63,15 +61,34 @@ plan(multisession, workers = 3)
...
+           geometry = geometries)
+### heat days 
+getHeat(folderPath = "data/raw/heatDays",
+        geometryLayers = geometries)
+
+
+# sensitive populations ----
+### asthma 
+getAsthma(filePath = "data/raw/asthma/co_asthma_hospitalization_nosupp_1822.csv",
```
[the remark this answers is not in the document]
Comment (dcarver1) [reply 0_main.R:84]: good note, I had to to that transformation outside of R as there were some issues reading in the xls file. 

Not perfect but we will deliver the final data with the csv format.

**Pull request merged**